core/core_modules/kpm.js

```diff
@@ -220,67 +220,104 @@ var git = require.once('../git.js'),
+                    fs.emptyDir(descriptor.folderPath, function () {
+                        // just delete if we can, not a lot we can do about errors here.
+                    });
+                }
+                cleanup();
+            });
+        }
+        catch (e) {
+            console.critical(e);
+            api.sendMessage('Could not install "' + name + '".', event.thread_id);
+            cleanup();
+        }
+    },
+
+    gitInstall = function(url, api, event) {
         api.sendMessage('Attempting to install module from "' + url + '"...', event.thread_id);
         tmp.dir(function (err, dir, cleanupCallback) {
             if (err) throw err;
-
             var cleanup = function(){
-                    fs.emptyDir(dir, function (err) {
-                        cleanupCallback(); // not a lot we can do about errors here.
-                    });
-                }.bind(this);
-
-            git.clone(url, dir, function(err, consoleOutput) {
-                try {
-                    var kj = require.once(path.join(dir, 'kassy.json')),
-                        moduleList = getModuleList();
-                    kj.safeName = sanitize(kj.name);
-                    if (this.loadedModules[kj.name] || moduleList[kj.name] || moduleList['kpm_' + kj.name]) {
-                        api.sendMessage('Module with name or directory "' + kj.name + '" has already been installed.', event.thread_id);
-                        cleanup();
-                        return;
-                    }
+                fs.emptyDir(dir, function () {
+                    cleanupCallback(); // not a lot we can do about errors here.
+                });
+            }.bind(this);
 
-                    if (!modules.verifyModuleDescriptior(kj)) {
-                        api.sendMessage('The repository at "' + url + '" is not a valid Kassy module.', event.thread_id);
-                        cleanup();
-                        return;
-                    }
+            git.clone(url, dir, function (err1) {
+                if (err1) {
+                    console.critical(err1);
+                    cleanup();
+                    return api.sendMessage('Failed to install module from "' + url + '"...', event.thread_id);
+                }
+                var parsed = urll.parse(url),
+                    cleaned = sanitize(path.basename(parsed.pathname));
+                return installCommon(cleaned, dir, cleanup, api, event);
+            });
+        }.bind(this));
+    },
 
-                    var instDir = path.resolve('./modules/kpm_' + kj.safeName);
-                    fs.copy(dir, instDir, function (err) {
-                        if (err) {
-                            console.debug(err);
-                            api.sendMessage('An unknown error occurred while installing "' + kj.name + '".', event.thread_id);
-                            cleanup();
-                            return;
-                        }
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ["scriptInstall" is not defined.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529976242/issues/source?bid=3282987&fileBranchId=3282998#l296)